crates/sui-framework/packages/bridge/sources/committee.move

```diff
@@ -98,18 +127,122 @@ module bridge::committee {
             if (!member.blocklisted) {
-                threshold = threshold + member.voting_power;
+                threshold = threshold + member.stake_amount;
             };
             i = i + 1;
             vec_set::insert(&mut seen_pub_key, pubkey);
         };
-        assert!(threshold >= required_threshold, ESignatureBelowThreshold);
+        assert!(threshold >= required_stake, ESignatureBelowThreshold);
     }
 
+    public(friend) fun register(
+        self: &mut BridgeCommittee,
+        system_state: &SuiSystemState,
+        bridge_pubkey_bytes: vector<u8>,
+        http_rest_url: vector<u8>,
+        ctx: &TxContext
+    ) {
+        // sender must be the same sender that created the validator object
+        let sender = tx_context::sender(ctx);
+        let validators = sui_system::validators(system_state);
+        let active_validators = validator_set::active_validators(validators);
+        let validator_index = find_validator(active_validators, sender);
+
+        assert!(option::is_some(&validator_index), ESenderNotActiveValidator);
+        // Sender is active validator, record the registration
+
+        // In case validator need to update the info
+        if (vec_map::contains(&self.member_registration, &sender)) {
+            let registration = vec_map::get_mut(&mut self.member_registration, &sender);
+            registration.http_rest_url = http_rest_url;
+            registration.bridge_pubkey_bytes = bridge_pubkey_bytes;
+        }else {
+            let registration = CommitteeMemberRegistration {
+                sui_address: sender,
+                bridge_pubkey_bytes,
+                http_rest_url,
+            };
+            vec_map::insert(&mut self.member_registration, sender, registration);
+        }
+    }
+
+    // This method will try to create the next committee using the registration and system state,
+    // if the total stake fails to meet the minimum required percentage, it will skip the update.
+    // This is to ensure we don't fail the end of epoch transaction.
+    public(friend) fun try_create_next_committee(
+        self: &mut BridgeCommittee,
+        system_state: &SuiSystemState,
+        min_stake_participation_percentage: u8,
+    ) {
+        let validators = sui_system::validators(system_state);
+        let active_validators = validator_set::active_validators(validators);
+        let total_member_stake = 0;
+        let i = 0;
+
+        let new_members = vec_map::empty();
+
+        while (i < vec_map::size(&self.member_registration)) {
+            // retrieve registration
+            let (_, registration) = vec_map::get_entry_by_idx(&self.member_registration, i);
+            // Find validator info from system state
+            let validator_index = find_validator(active_validators, registration.sui_address);
+            // Process registration if it's active validator
+            if (option::is_some(&validator_index)) {
```

Comment:
when we create this committee a certain, if validator became inactive, we would just ignore it?

Reply:
yes, the validator will need to be active at the time of committee creation